module/VuFind/src/VuFind/View/Helper/Bootstrap3/Captcha.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Captcha view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <crhallberg@gmail.com>
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Bootstrap3;
+
+/**
+ * Captcha view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <crhallberg@gmail.com>
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Captcha extends \VuFind\View\Helper\Root\Captcha
+{
+    /**
+     * Generate HTML depending on CAPTCHA type (empty if not active).
+     *
+     * @param bool $useCaptcha Boolean of active state, for compact templating
+     * @param bool $wrapHtml   Wrap in a form-group?
+     *
+     * @return string
+     */
+    public function html(bool $useCaptcha = true, bool $wrapHtml = true): string
```

Comment:
This seems significantly different from the parent class' `html()` method. I wonder if this should be refactored into smaller parts (or made to use class properties) so that it's easier to separate the styling logic (which will vary from theme to theme) from the actual functional logic (which should be made consistent). Or perhaps the method should be made abstract in the root class for now, until we have another use case that justifies the refactoring.

Reply:
Since whatever we create might not be the correct solution for all future use cases, i just made the Root class abstract and hope that it's fine this way

Reply:
OK, so i created a Helpers/captcha.phtml file which works fine. Regarding AbstractClassBasedTemplateRenderer, all other classes which use that are View Helpers so far. If i understand correctly, your proposal is that VuFind\Captcha\AbstractBase should inherit from AbstractClassBasedTemplateRenderer. I will give it  a try!

Reply:
Actually, I wasn't suggesting that VuFind\Captcha\AbstractBase should extend AbstractClassBasedTemplateRenderer; rather, I was suggesting that the Captcha view helper should. Then, the view helper can look at each Captcha object it is passed, and select an appropriate template to render based on the object's class name.

Reply:
OK, now this works, but i had to add several getters in the VuFind\Captcha\* classes to get type-specific variables when needed in the templates.

Reply:
Great, that's exactly what I had in mind! Thanks for doing the refactoring.